### DISCUSSION TITLE
[2022/01/04] Frontend Night

### DISCUSSION THREAD

**Discussion by jjenko** [wantedly/frontend-night#186] [Archive]
## Policy

- 資料の準備をがんばらない
- 参加・途中退席は自由
- スキップしない

話したい・聞きたいネタを書いてく ✏️

ハッシュタグ [#frontend_night](https://twitter.com/hashtag/frontend_night?src=hash)

Prev: https://github.com/wantedly/frontend-night/discussions/185 

 _Created from https://github.com/wantedly/frontend-night/discussions/179 by [issue-creator](https://github.com/rerost/issue-creator)_

**Comment by kobayang** [comment]
## Supabase

https://supabase.com/

- Firebase Alternative を掲げている。
- 誤解を恐れずに言えば Firebase の RDB版。(Firestore → Postgres)

**Reply by kobayang** [reply]
## 面白いポイント

- Postgres の機能をフルに使っているところ
  - e.g. [Row Level Security (RLS)](https://www.postgresql.org/docs/current/ddl-rowsecurity.html) と [Policies](https://www.postgresql.org/docs/current/sql-createpolicy.html) で認可のロジックをかく
- Firebase に対応した機能をそれぞれOSSを使って実現している

![image](https://user-images.githubusercontent.com/12181179/148019931-aef96c31-e89c-40a0-b80c-02a781ea5b0d.png)

c.f. https://supabase.com/docs/architecture

**Reply by kobayang** [reply]
## PostgREST 

supabase は以下のように（あたかもDBを直接 select するように）データを取り出せる。

```tsx
const { data, error } = await supabase
  .from('cities')
  .select()
```

内部では、 Postgres を rest api 経由で透過的に扱える [Postgrest](https://github.com/PostgREST/postgrest) の API があって、
 [postgrest-js](https://github.com/supabase/postgrest-js) Client を supabase 経由で使っている。


色々触ったけど、複雑なことをしようとすると難しい印象だった。
（inner join + 絞り込みとかもできないことはなさそうだけど難しい）

**Reply by kobayang** [reply]
Postgres + Rest API があるんだったら Postgres + GraphQL API もある
https://www.graphile.org/postgraphile/
